x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/RestSqlQueryAction.java

```diff
@@ -45,8 +49,18 @@ public RestSqlQueryAction(Settings settings, RestController controller) {
     protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
         SqlQueryRequest sqlRequest;
         try (XContentParser parser = request.contentOrSourceParamParser()) {
-            sqlRequest = SqlQueryRequest.fromXContent(parser, Mode.fromString(request.param("mode")),
-                                                      RestClient.fromString(request.param("client")));
+            String clientId = request.param("clientid");
+            if (clientId != null) {
+                if (clientId.length() > CLIENT_ID_MAX_LENGTH) {
+                    clientId = clientId.substring(0, CLIENT_ID_MAX_LENGTH).toLowerCase(Locale.ROOT);
+                }
+                if (!clientId.equals(CLI) && !clientId.equals(CANVAS)) {
```

Comment:
Not sure what this block tries to achieve. If just `CLI` and `CANVAS` are accepted, why bother checking the length?